owsi-core/owsi-core-components/owsi-core-component-jpa/src/main/java/fr/openwide/core/jpa/business/generic/model/GenericEntity.java

```diff
@@ -127,12 +128,7 @@ public boolean equals(Object object) {
 
 	@Override
 	public int hashCode() {
-		int hash = 7;
-		
-		K id = getId();
-		hash = 31 * hash + ((id == null) ? 0 : id.hashCode());
-
-		return hash;
+		return System.identityHashCode(HibernateUtils.unwrap(this));
```

Comment:
Right. It seems like proxies override `hashCode` anyway, so this particular implementation will never be called on a proxy.